### PR TITLE
MM-25931 Handle incoming session expired push notification

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR handles incoming remote push notifications for session expired.

Needs to be tested with Related PRs:
*  https://github.com/mattermost/mattermost-push-proxy/pull/67
* https://github.com/mattermost/mattermost-server/pull/14732

Previous app versions:
* Android no notification visible
* iOS notification displays only when the app is either closed or in the background.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-25931

### CODE REVIEW

File: ios/Mattermost/AppDelegate.m [hunk 1]
```diff
@@ -71,24 +71,18 @@ -(void)application:(UIApplication *)application didReceiveRemoteNotification:(no
   NSString* action = [userInfo objectForKey:@"type"];
   NSString* channelId = [userInfo objectForKey:@"channel_id"];
   NSString* ackId = [userInfo objectForKey:@"ack_id"];
+  RuntimeUtils *utils = [[RuntimeUtils alloc] init];
 
-  if (action && [action isEqualToString: NOTIFICATION_CLEAR_ACTION]) {
+  if ((action && [action isEqualToString: NOTIFICATION_CLEAR_ACTION]) || (state == UIApplicationStateInactive)) {
     // If received a notification that a channel was read, remove all notifications from that channel (only with app in foreground/background)
     [self cleanNotificationsFromChannel:channelId];
```
Comment (wiggin77) [inline ios/Mattermost/AppDelegate.m:78]: Is this safe with an empty channeId?

Comment (enahum) [reply ios/Mattermost/AppDelegate.m:78]: This method does not get called for session expired notifications, only for clear notifications cause of what the payload contains

File: ios/Mattermost/AppDelegate.m [hunk 2]
```diff
@@ -71,24 +71,18 @@ -(void)application:(UIApplication *)application didReceiveRemoteNotification:(no
   NSString* action = [userInfo objectForKey:@"type"];
   NSString* channelId = [userInfo objectForKey:@"channel_id"];
   NSString* ackId = [userInfo objectForKey:@"ack_id"];
+  RuntimeUtils *utils = [[RuntimeUtils alloc] init];
 
-  if (action && [action isEqualToString: NOTIFICATION_CLEAR_ACTION]) {
```
Comment (agnivade) [inline ios/Mattermost/AppDelegate.m:75]: Not well-versed with this codebase, so perhaps a silly question: is this refactoring related to this PR or just a general thing ?

Comment (enahum) [reply ios/Mattermost/AppDelegate.m:75]: General, while looking at this I saw the there was unnecessary duplicate code

**Comment by enahum** [comment]
manual cherry-pick
